src/solver/expressions/visitors/AstDOTStyleVisitor.cpp

```diff
@@ -52,34 +64,20 @@ void AddFiliation(std::ostream& os, const std::string& parent_id, const std::str
     os << "legend_" << parent_id << " -> " << "legend_" << child_id << " [style=invis];\n";
 }
 
-void GetLegend(const std::map<std::string, std::map<const Nodes::Node*, unsigned int>>& nodeIds,
-               std::ostream& os)
+void GetLegend(std::map<std::string, unsigned int>& nbNodesByType, std::ostream& os)
```

Comment:
```suggestion
void GetLegend(const std::map<std::string, unsigned int>& nbNodesByType, std::ostream& os)
```

Reply:
Changed : **GetLegend** is now **makeLegend** and is a member function taking one argument `std::ostream& os`